peachjam/models/citations.py

```diff
@@ -78,3 +84,36 @@ def for_target_works(cls, work):
             .filter(target_work=work)
             .order_by("citing_work__title")
         )
+
+
+class CitationProcessing(SingletonModel):
+    processing_date = models.DateField(_("processing date"), null=True, blank=True)
+
+    def update_processing_date(self, date):
+        self.processing_date = date
+        self.save()
+
+    def re_extract_citations(self):
+        """Extract citations for all documents dated on or after the processing date."""
+        from peachjam.models import CoreDocument
+        from peachjam.tasks import extract_citations
+
+        later_documents = CoreDocument.objects.filter(
+            date__gte=self.processing_date
+        ).order_by("date")
+        log.info("Re-extracting citations for %s documents", later_documents.count())
```

Comment:
Also include the date in the log message, which will help us debug it if necessary.